api/http/management.go

```diff
@@ -360,7 +374,12 @@ func (h ManagementController) websocketWriter(
 
 	recorderBuffered := bufio.NewWriterSize(recorder, app.RecorderBufferSize)
 	defer recorderBuffered.Flush()
+	controlRecorderBuffered := bufio.NewWriterSize(controlRecorder, app.RecorderBufferSize)
+	defer controlRecorderBuffered.Flush()
 	recordedBytes := 0
+	controlBytes := 0
+	lastKeystrokeAt := int64(float64(time.Now().UTC().UnixNano()) * 0.000001)
+	//controlRecordedBytes := 0
```

Comment:
These buffers and variables are allocated unconditionally of the underlying protocol.

Reply:
I know, but the only option at the moment is to check if they are allocated each time `ProtoTypeShell` message comes (in this place we have no idea what protocol is going to be used), and a couple of ifs on *every* byte transferred is much worse than two small buffers allocated and not used.

Reply:
in other words: I am trying to convince you that memory usage will be neglectable, since I hope go lang will resize the regions if needed, and not allocate 15Mb at the start.